resources/prosody-plugins/mod_auth_token.lua

```diff
@@ -102,6 +112,11 @@ function provider.get_sasl_handler(session)
             self.username = message;
         end
 
+        local event_result = prosody.events.fire_event("post-jitsi-authentication", session);
+        if event_result ~= nil then
+            validate_result(session,event_result.res, event_result.error, event_result.reason)
```

Comment:
I don't believe that event_result is going to end up like you expect here, instead this should look like: `res, error, reason = prosody.events.fire_event`

Reply:
From fire_event you can return either a value or an object, so where I hook to this event I will pack the result in the appropriate format.